asto-core/src/main/java/com/artipie/asto/Key.java

```diff
@@ -53,11 +53,39 @@ public interface Key {
      */
     List<String> parts();
 
+    /**
+     * Base key class.
+     * @since 1.14.0
+     */
+    abstract class BaseKey implements Key {
```

Comment:
I'd suggest to call this class just `Base` without `key` as this class is a part of `Key` interface and will always be accessed as `Key.Base`.

Reply:
ok